apps/question-bank/src/main/java/gov/cdc/nbs/questionbank/valueset/ValueSetController.java

```diff
@@ -35,71 +32,71 @@
-        return valueSetReador.searchValueSearch(search, pageable);
-    }
-
-    @GetMapping("{codeSetNm}/concepts")
-    public List<Concept> findConceptsByCodeSetName(@PathVariable String codeSetNm) {
-        return valueSetReador.findConceptCodes(codeSetNm);
-    }
-
-    @PutMapping("/{codeSetNm}/concepts/{conceptCode}")
-    public Concept updateConcept(
-            @PathVariable String codeSetNm,
-            @PathVariable String conceptCode,
-            @RequestBody UpdateConceptRequest request,
-            @ApiIgnore @AuthenticationPrincipal final NbsUserDetails details) {
-        return conceptUpdater.update(codeSetNm, conceptCode, request, details.getId());
-    }
-
-    @PostMapping("{codeSetNm}/concepts")
-    public Concept addConcept(@PathVariable String codeSetNm, @RequestBody AddConceptRequest request) {
-        Long userId = userDetailsProvider.getCurrentUserDetails().getId();
-        return conceptCreator.addConcept(codeSetNm, request, userId);
-    }
+  private final ValueSetStateManager valueSetStateManager;
+  private final ValueSetReader valueSetReader;
+  private final ValueSetUpdater valueSetUpdater;
+  private final ValueSetCreator valueSetCreator;
+  private final ConceptCreator conceptCreator;
+  private final ConceptUpdater conceptUpdater;
+  private final UserDetailsProvider userDetailsProvider;
+
+  @PostMapping
+  public ResponseEntity<CreateValueSetResponse> createValueSet(@RequestBody ValueSetRequest request) {
+    Long userId = userDetailsProvider.getCurrentUserDetails().getId();
+    CreateValueSetResponse response = valueSetCreator.createValueSet(request, userId);
+    return new ResponseEntity<>(response, null, response.getStatus());
+
+  }
+
+  @PutMapping("{codeSetNm}/deactivate")
+  public ResponseEntity<ValueSetStateChangeResponse> deleteValueSet(@PathVariable String codeSetNm) {
+    ValueSetStateChangeResponse response = valueSetStateManager.deleteValueSet(codeSetNm);
+    return new ResponseEntity<>(response, null, response.getStatus());
+  }
+
+  @PutMapping("{codeSetNm}/activate")
+  public ResponseEntity<ValueSetStateChangeResponse> activateValueSet(@PathVariable String codeSetNm) {
+    ValueSetStateChangeResponse response = valueSetStateManager.activateValueSet(codeSetNm);
+    return new ResponseEntity<>(response, null, response.getStatus());
+  }
+
+  @GetMapping
+  public Page<ValueSet> findAllValueSets(@PageableDefault(size = 25) Pageable pageable) {
+    return valueSetReader.findAllValueSets(pageable);
+
+  }
+
+  @PostMapping("/update")
+  public ResponseEntity<UpdatedValueSetResponse> updateValueSet(@RequestBody ValueSetUpdateRequest update) {
+    UpdatedValueSetResponse response = valueSetUpdater.updateValueSet(update);
+    return new ResponseEntity<>(response, null, response.getStatus());
+
+  }
+
+  @PostMapping("search")
+  public List<ValueSetSearchResponse> searchValueSet( @PageableDefault(size = 25) Pageable pageable
```

Comment:
Suggestion: Per Lakshmi's comment, to provide paging information this can be changed to return a `Page<ValueSetSearchResponse>`